05-JWT-Basics/preferred/middleware/authentication.js

```diff
@@ -0,0 +1,24 @@
+const jwt = require("jsonwebtoken");
+
+const authentication = (req, res, next) => {
+  const authHeader = req.headers.authorization;
+
+  if (!authHeader) {
+    return res.status(401).json({ message: "Unauthorized" });
+  }
+
+  const token = authHeader.split(" ")[1];
+
+  try {
+    const secret = process.env.SECRET;
+    const decoded = jwt.verify(token, secret);
+
+    req.user = { userName: decoded.userName };
+
+    next();
+  } catch (err) {
+    return res.status(401).json({ message: "Unauthorized" });
+  }
+};
```

Comment:
Great error handling in this method. If you wanted to simplify even further, you could use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) to access the auth header and let the final catch throw the only error.

```javascript
const authHeader = req.headers.authorization
const token = authHeader?.split(" ")[1]; // This will resolve to undefined if `authHeader` is undefined.
```